### PR TITLE
Using sync api

### PR DESCRIPTION
**Pull request by harshalpatel91**
For Code review

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,9 +21,9 @@ plugins: [
       // API Key is a unique key assigned to each stack. This is required.
       api_key: `api_key`,
 
-      // Access Token is a read-only credential . This is required.
-      access_token: `access_token`,
-
+      // Delivery Token is a read-only credential . This is required.
+      deliver_token: `deliver_token`,
```
Comment (DSchau) [inline README.md]: This is usually called `delivery_token`. Is this correct?

File: gatsby-node.js [hunk 2]
```diff
@@ -4,60 +4,179 @@ var _regenerator = require("babel-runtime/regenerator");
 
 var _regenerator2 = _interopRequireDefault(_regenerator);
```
Comment (DSchau) [inline gatsby-node.js]: Don't want to publish your built files! May want to add this to your gitignore!

File: src/gatsby-node.js [hunk 3]
```diff
@@ -1,35 +1,132 @@
...
+        store.getState().status.plugins[`gatsby-source-contentstack`][
+            `contentstack-sync-token-${configOptions.api_key}`
+        ]
+    ) {
+        syncToken = store.getState().status.plugins[`gatsby-source-contentstack`][
+            `contentstack-sync-token-${configOptions.api_key}`
+        ];
+    }
 
+    configOptions.syncToken = syncToken || null;
```
Comment (DSchau) [inline src/gatsby-node.js]: Would this typically be passed in via the config? Seems like this should be something computed by the plugin, right?

File: src/gatsby-node.js [hunk 4]
```diff
@@ -1,35 +1,132 @@
...
-
-        entries.forEach(entry => {
-            const normalizedEntry = normalizeEntry(contentType, entry, contentstackData.entries, createNodeId);
-            // Process the contentTypes data to match the structure of a Gatsby node
-            const entryNode = processEntry(contentType, normalizedEntry, createNodeId);
-            // Use Gatsby's createNode helper to create a node from the node data
-            createNode(entryNode);
+    const { contentstackData } = await fetchData(configOptions);
+    
+    let publishedEntriesType   = contentstackData.syncData.filter(item => item.type === 'entry_published') || [];
```
Comment (DSchau) [inline src/gatsby-node.js]: Why not do this grouping once? e.g.

```js
const data = contentstackData.syncData.reduce((merged, item) => {
  if (!merged[item.type]) {
    merged[item.type] = []
  }
  merged[item.type].push(item)
  return merged
}, {})
```

then you can do data['entry_unpublished'] to get unpublished (and other types)

File: src/normalize.js [hunk 5]
```diff
@@ -22,6 +22,26 @@ exports.processContentType = (content_type, createNodeId) => {
     return nodeData;
 }
 
+exports.processAsset = (asset, createNodeId) => {
+    const nodeId = makeAssetNodeUid(asset, createNodeId);
+    const nodeContent = JSON.stringify(asset);
+    const nodeContentDigest = crypto
```
Comment (DSchau) [inline src/normalize.js]: We pass this exact function as `createContentDigest` to sourceNodes. To grab it:

```js
exports.sourceNodes = ({ actions, createContentDigest }) => {
  // use or pass createContentDigest
}
```

File: gatsby-node.js (hunk 2 above)
Comment (DSchau) [reply gatsby-node.js]: Just make sure you add an npmignore that only ignores the src directory!

File: README.md (hunk 1 above)
Comment (harshalpatel91) [reply README.md]: Yes, It was a typo

File: gatsby-node.js (hunk 2 above)
Comment (harshalpatel91) [reply gatsby-node.js]: Yes, will add a npmignore file here

File: src/gatsby-node.js [hunk 6]
```diff
@@ -1,35 +1,132 @@
...
 
-exports.sourceNodes = async ({ boundActionCreators, createNodeId },configOptions) => {
-    const { createNode } = boundActionCreators;
 
-    const { contentstackData } = await fetchData(configOptions);
+exports.sourceNodes = async ({ actions, getNode, getNodes, createNodeId, store, reporter },configOptions) => {
+    const { createNode, deleteNode, touchNode, setPluginStatus } = actions;
+    let syncToken 
+
+    if (
```
[the remark this answers is not in the document]
Comment (harshalpatel91) [reply src/gatsby-node.js]: Done

File: src/gatsby-node.js (hunk 3 above)
Comment (harshalpatel91) [reply src/gatsby-node.js]: We have computed and just added on configOptions obj and passed in fetch method.

File: src/gatsby-node.js (hunk 4 above)
Comment (harshalpatel91) [reply src/gatsby-node.js]: Done

File: src/normalize.js (hunk 5 above)
Comment (harshalpatel91) [reply src/normalize.js]: Done

**Pull request merged**